src/Relude/List.hs

```diff
@@ -61,6 +62,21 @@ infix 9 !!?
     go _ []     = Nothing
 {-# INLINE (!!?) #-}
 
+{- | Partitions a list based on the result of function which produces an Either value. List of all elements producing Left are extracted, in order, to the first element of the output tuple. Similarly, a list of all elements producing Right are extracted to the second element of output.
+
+>>> :{
+ foo x
+   | even x = Left x
+   | otherwise = Right x
+ :}
+
+>>> partitionWith foo [1 .. 10]
+([2,4,6,8,10],[1,3,5,7,9])
+-}
+partitionWith :: (a -> Either b c) -> [a] -> ([b], [c])
+partitionWith = (partitionEithers .) . map
```

Comment:
Let's write slightly more pointful implementation. It will be more readable and maintainable in long-term.
```suggestion
partitionWith f = partitionEithers . map f
```